### ISSUE TITLE
Stage hunks of untracked file

### ISSUE THREAD

**Issue by jrwrigh** [tpope/vim-fugitive#1683]
Currently, if you open the diff of an untracked file and try to stage hunks (ie. in visual mode using `diffput`), it returns with an error:

`E793: No other buffer in diff mode is modifiable`.

It'd be really nice to have the empty buffer be writable so that for an untracked file, you could selectively stage specific parts of the file.

## Edit:

I've realized there is an additional requirement to reproduce the issue. The untracked file must be inside an untracked directory. ie. if you have an untracked `example/example.sh`, then `:G` shows the following:

```
Untracked (1)
? example/
```
Thus you can't run `dd` on the file itself.

If you go to stage the file, then we get this:
```
Staged (1)
A example/example.sh
```

Now we can run `dd` on the file. However, then we run into the above problem when trying to stage hunks.

**Comment by tpope** [comment]
Can't reproduce. What is `:echo @%` in the newly opened diff window?

**Comment by jrwrigh** [comment]
Realized there was an additional step to reproduce the problem. I've added the additional information to the original message.

tl;dr. The untracked file must be within an untracked directory.

**Comment by jrwrigh** [comment]
`:echo @%` for the "empty" buffer (ie. the one to write to for staging hunks):
`fugitive:///users/jrwrigh/gitRepos/vtkpytools/.git//dd87476f56e4a01f0801903688a8bf89bf033ec6/tests/examples/test_examples.py`

`:echo @%` for the "full" buffer (ie. the one containing the new file)r:
`fugitive:///users/jrwrigh/gitRepos/vtkpytools/.git//0/tests/examples/test_examples.py`

**Comment by tpope** [comment]
If you call `dd` on a staged file, you get a diff between what's going to be committed and what's in the most recent commit. The behavior is correct, and I had to sum up your problem, it would be "your untracked file isn't untracked at all."

You might be interested in `git add --intent-to-add`, which is available in Fugitive on `I`.

**Comment by jrwrigh** [comment]
> I had to sum up your problem, it would be "your untracked file isn't untracked at all."
That's not the title of the issue. That's a description of why I get an error message when I attempt to solve the issue.

> You might be interested in git add --intent-to-add, which is available in Fugitive on I.
That's the solution to the actual problem. Thanks!
